### PR TITLE
Added setup for automation and general improvements

### PR DESCRIPTION
**Pull request by Samsquamptch**
After running !setup, the bot can create channels and roles as required before saving the data to a yaml config file. This file allows for the bot to load the admin and queue embeds/buttons automatically on startup, avoiding the need for commands to load these. I've also set up the requirements.txt file and a few other refactoring improvements. Menus no longer require the "refresh" option to do the same thing twice in a row, for example.

To be honest, after merging this with dev, the bot will be viable from a basic user perspective. Yes, you will need to manually clear the queue once a game is over using kick and none of the data will be stored, but people can sign up and have teams balanced for them. We could run a user test over a coming weekend for feedback on the discord side of things and see what people think.

### CODE REVIEW

File: .gitignore
Comment (Teky500) [inline .gitignore:1]: I think this file is needed to clearly show which files are to be ignored/not in source control.

File: credentials/.gitignore
Comment (Teky500) [inline credentials/.gitignore:1]: This file should basically be:
*
!.gitignore

File: .gitignore
Comment (Teky500) [reply .gitignore:1]: Perhaps files like data/users.csv should be here.

**Comment by Samsquamptch** [comment]
I've tried to recreate the gitignore, not sure what to do now.

File: .gitignore
Comment (Samsquamptch) [reply .gitignore:1]: fixed

**Review by Teky500 — APPROVED** [review]

**Pull request merged**